daisy_workflows/image_build/debian/build_fai.py

```diff
@@ -49,6 +49,71 @@ def mycopytree(src, dst):
     return dst
 
 
+# get number of bytes to offset when mounting the raw disk image file
+def get_mount_offset(fdisk_output):
+    single_str = fdisk_output.stdout.decode('utf-8')
+    subprocess.run(['echo', 'single string'])
+    subprocess.run(['echo', single_str])
+    lines = single_str.splitlines()
+    subprocess.run(['echo', 'number of lines: ' + str(len(lines))])
+    for line in lines:
+      subprocess.run(['echo', str(line)])
+
+    device_title_line = -1
+
+    offset_scaling_factor = 1
+
+    # the index in the title line where the word 'start' is
+    # ['Device', 'Start', 'End', 'Sectors', 'Size', 'Type']
+    # For this line, the index would be 1.
+    start_index = -1
+
+    # Same as start_index, but for the word 'type'
+    type_index = -1
+
+    # offset as the input to loop
+    overall_offset = 0
+    # true if the next line is expected to be a device
+    for i in range(len(lines)):
+        line = lines[i]
+        subprocess.run(['echo', 'SBOMline'])
+        subprocess.run(['echo', line])
+        tokens = str.split(line)
+        if len(tokens) > 0:
+            if tokens[0].lower() == 'units:':
+                bytes_index = tokens.index('bytes')
+                offset_scaling_factor = int(tokens[bytes_index - 1])
+            if tokens[0].lower() == 'device':
+                device_title_line = i
+                start_index = tokens.index('Start')
+                type_index = tokens.index('Type')
+    for i in range(device_title_line + 1, len(lines)):
+        line = lines[i]
+        tokens = str.split(line)
+        if len(tokens) == 0:
+            break
+        if tokens[type_index].lower() == 'linux' and \
+           tokens[type_index + 1].lower() == 'filesystem':
+            overall_offset = offset_scaling_factor * int(tokens[start_index])
+    return overall_offset
+
+
+# Syft SBOM generation steps
+def run_sbom_generation(syft_source, disk_name, work_dir, offset):
+    subprocess.run(['gsutil', 'cp', syft_source, 'syft.tar.gz'])
+    subprocess.run(['tar', '-xf', 'syft.tar.gz'])
```

Comment:
Please check for the command success everywhere.

Reply:
added check=True to the commands